integrations/opsgenie/integration.py

```diff
@@ -0,0 +1,109 @@
+from typing import Any, Literal
+from pydantic import Field, BaseModel
+
+from port_ocean.core.handlers import APIPortAppConfig
+from port_ocean.core.handlers.port_app_config.models import (
+    ResourceConfig,
+    PortAppConfig,
+    Selector,
+)
+from port_ocean.core.integrations.base import BaseIntegration
+
+
+class APIQueryParams(BaseModel):
+    created_at: str | None = Field(
+        alias="createdAt", description="The date and time the alert was created"
+    )
+    last_occurred_at: str | None = Field(
+        alias="lastOccurredAt",
+        description="The date and time the alert was last occurred",
+    )
+    snoozed_until: str | None = Field(
+        alias="snoozedUntil",
+        description="The date and time the alert was snoozed until",
+    )
+    message: str | None = Field(description="The message of the alert or incident")
+    status: Literal["open", "resolved", "closed"] | None = Field(
+        description="The status of the alert"
+    )
+    is_seen: Literal["true", "false"] | None = Field(
+        description="Whether the alert has been seen"
+    )
+    acknowledged: Literal["true", "false"] | None = Field(
+        description="Whether the alert has been acknowledged"
+    )
+    snoozed: Literal["true", "false"] | None = Field(
+        description="Whether the alert has been snoozed"
+    )
+    priority: Literal["P1", "P2", "P3", "P4", "P5"] | None = Field(
```

Comment:
is there possibility to specify your own priorities in opsgenie? I wonder if we actually want to restrict it, as it means any addition on opsgenie side, will require us to update the code